packages/node/src/transport/HttpInboundTransport.ts

```diff
@@ -52,10 +60,24 @@ export class HttpInboundTransport implements InboundTransport {
       try {
         const message = req.body
         const encryptedMessage = JSON.parse(message)
-        await messageReceiver.receiveMessage(encryptedMessage, {
-          session,
+        agent.events.emit<AgentMessageReceivedEvent>(agent.context, {
+          type: AgentEventTypes.AgentMessageReceived,
+          payload: {
+            message: encryptedMessage,
+            session: session,
+          },
         })
 
+        // Wait for message to be processed
+        await firstValueFrom(
+          agent.events.observable<AgentMessageProcessedEvent>(AgentEventTypes.AgentMessageProcessed).pipe(
+            filter((e) => e.type === AgentEventTypes.AgentMessageProcessed),
+            filter((e) => e.payload.message.id === encryptedMessage.id),
+            filter((e) => e.payload.message.type === encryptedMessage.type),
+            first()
+          )
+        )
```

Comment:
If the message is processed very quickly, this won't work, we need to start the event listener before. Also if processing somehow fails this will never emit. Do we have the right guardrails in place that this event will always be emitted? Otherwise we may want to add a timeout?

Reply:
I have added a timeout of `10` seconds.